en/dev_log/ulog_file_format.md

```diff
@@ -41,233 +48,318 @@ The file consists of three sections:
-  A field has the format: `type field_name` or `type[array_length] field_name` for arrays (only fixed size arrays are supported).
-  `type` is one of the basic binary types or a `message_name` of another format definition (nested usage).
-  A type can be used before it's defined.
-  There can be arbitrary nesting but no circular dependencies.
-
-  Some field names are special:
-  - `timestamp`: every logged message (`message_add_logged_s`) must include a timestamp field (does not need to be the first field).
-    Its type can be: `uint64_t` (currently the only one used), `uint32_t`, `uint16_t` or `uint8_t`.
-    The unit is always microseconds, except for in `uint8_t` it's milliseconds.
-    A log writer must make sure to log messages often enough to be able to detect wrap-arounds and a log reader must handle wrap-arounds (and take into account dropouts).
-    The timestamp must always be monotonic increasing for a message series with the same `msg_id`.
-  - Padding: field names that start with `_padding` should not be displayed and their data must be ignored by a reader.
-    These fields can be inserted by a writer to ensure correct alignment.
-
-    If the padding field is the last field, then this field will not be logged, to avoid writing unnecessary data.
-    This means the `message_data_s.data` will be shorter by the size of the padding.
-    However the padding is still needed when the message is used in a nested definition.
-
-- 'I': information message.
-  ```c
-  struct message_info_s {
-  	struct message_header_s header;
-  	uint8_t key_len;
-  	char key[key_len];
-  	char value[header.msg_size-1-key_len]
-  };
-  ```
-  `key` is a plain string, as in the format message (can also be a custom type), but consists of only a single field without ending `;`, eg. `float[3] myvalues`.
-  `value` contains the data as described by `key`.
+:::note
+Message types described below are noted with the single alphabet `msg_type` characteristic in the beginning.
+:::
+
+
+### Definitions Section
+
+The definitions section contains basic information such as software version, uORB Topic format, initial parameter values, and so on.
+
+The message types in this section are:
+1. [Flag Bits](#b--flag-bits-message)
+2. [Format Definition](#f--format-definition-message)
+3. [Information](#i--information-message)
+4. [Multi Information](#m--multi-information-message)
+5. [Parameter](#p--parameter-message)
+6. [Default Parameter](#q--default-parameter-message)
+
+
+#### 'B': Flag Bits Message
+
+:::note
+This message must be the **first message** right after the header section, so that it has a fixed constant offset from the start of the file!
+:::
+
+```
+struct ulog_message_flag_bits_s {
+  struct message_header_s;
+  uint8_t compat_flags[8];
+  uint8_t incompat_flags[8];
+  uint64_t appended_offsets[3]; ///< file offset(s) for appended data if appending bit is set
+};
+```
+
+- `compat_flags`: compatible flag bits.
```

Comment:
@bkueng Is there a better way to phrase this? Since I am considering explaining this part like:
> Compatible flags indicate if the log includes any special message that is compatible with the existing parsers is in the uLog

And the `incompat_flags` part like:
> Incompatible flags indicate whether any special message that is incompatible with the existing parsers is in the uLog

But I am not sure if I understand this part correctly (Also out of 64 bits, only 1 bit is being used for now :thinking:), could you give your input on this?